src/Bicep.LangServer/Completions/ResourceTypeSearchKeywords.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Bicep.Core.Parsing;
+using Bicep.Core.Resources;
+
+namespace Bicep.LanguageServer.Completions
+{
+    public class ResourceTypeSearchKeywords
+    {
+        private ImmutableDictionary<string, string[]> keywordsMap;
```

Comment:
To make this fully immutable:
```suggestion
        private ImmutableDictionary<string, ImmutableArray<string>> keywordsMap;
```